ADR/ADR022-use-postgresql-for-form-creation-in-form-admin.md

```diff
@@ -0,0 +1,41 @@
+# ADR022:  (a description of the decision, not the problem)
+
+Date: 2023-09-14
+
+## Status
+
+> Proposed
+
+## Context
+
+Originally forms-admin didn't have a workflow but a single screen for the user to draft their form. This got broken up into a multistage workflow. Data is transferred between screens via cookies but this has limitations, and only worked ok for a linear flow. Now that we are approaching diverging workflows, we can't support this any longer. We are also currently only using ActiveModel for our classes.
+
+We should:
+
+* Refactor our models to be more Rails-standard and switch from ActiveModel only to ActiveRecord.
+
+* Use the PostgreSQL database to manage data.
+
+* Drop using cookies for data in session.
+
+See https://trello.com/c/oEOqJ4BU/1124-draft-refactor-forms-admin-to-use-activerecord-postgresql-for-form-drafting
+
+## Decision
+
+> what the team has decided to do
+
+## Consequences
+
+### Pros
+
+* removes need for Redis instance
+* it makes the app more standard Rails
+* it removes massive tech debt (where we went from a single-page, zero workflow interface to trying to adapt it to have some workflow)
+* it makes things easier to implement
+* it’s easier to test
+* data is more stable
+* users don’t lose their data as their workflow diverges
+* as more complexity is added to the workflow, we won’t be adding to that tech debt and working against and increasing brittleness
+
+### Cons
```

Comment:
Are there any cons at all? 😁 Do we need to do garbage-collection in PostgreSQL? Do we need to put other work on hold while doing this refactor? What about deploying, does that need some thought?

Reply:
This is why I'm drafting the ADR! I can't think of any... The main one might be that we start to duplicate some of forms-api locally, however I feel that's a cost worth taking and we can keep api as the more "source of truth" of forms for rendering and published state. It's possible it'll make it easier to merge the two, should we decide to do.

Reply:
I think one cons might be that we are "splintering" the data into much smaller chunks which would then have to be "pieced" together before sending data to forms-api.